sources/lib/frontend/typechecker.ml

```diff
@@ -37,7 +37,7 @@ module S = Set.Make(String)
 module HSS = Hstring.Set
 
 module MString =
-  Map.Make(struct type t = string let compare = Pervasives.compare end)
+  Map.Make(struct type t = string let compare = Stdlib.compare end)
```

Comment:
Here, using `String.compare` (or at least a string-specialised version of comparison) would probably be better I think.